src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  // died() {
+  //   if (this.health <= 0) {
+  //     delete Animal.alive.this;
+  //   }
+  // }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.name = name;
+  }
+
+  bite(name) {
+    if (name.hidden === false) {
+      name.health -= 50;
+    }
```

Comment:
The `bite` method should check if the target is an instance of `Herbivore` and not currently hiding. Additionally, it should not act on `Carnivore` instances. The current check only verifies if `hidden` is `false`, which is not sufficient to meet the requirements.